src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java

```diff
@@ -589,20 +590,29 @@ public FeatureCollectionResponse run(GetFeatureRequest request) throws WFSExcept
                                         .getConnectionParameters()
                                         .get(WFSDataStoreFactory.USEDEFAULTSRS.key)
                                         .toString())) {
-                            String otherSRS =
-                                    (String) meta.getMetadata().get(FeatureTypeInfo.OTHER_SRS);
+
                             if (query.getSrsName() != null) {
-                                if (otherSRS.contains(query.getSrsName().toString())) {
-                                    if (hints == null) hints = new Hints();
-                                    try {
-                                        hints.put(
-                                                ResourcePool.MAP_CRS,
-                                                CRS.decode(query.getSrsName().toString()));
-                                    } catch (NoSuchAuthorityCodeException ne) {
-                                        LOGGER.log(Level.SEVERE, ne.getMessage(), ne);
-                                    } catch (FactoryException e) {
-                                        LOGGER.log(Level.SEVERE, e.getMessage(), e);
+                                String otherSrsStr =
+                                        (String) meta.getMetadata().get(FeatureTypeInfo.OTHER_SRS);
+                                // create list of other srs
+                                List<String> otherSRSList = Arrays.asList(otherSrsStr.split(","));
```

Comment:
Won't this go NPE if otherSrsStr is null? Unless configured it will be missing and thus null no?

Reply:
Please check line 585, this block will not execute if otherSrsStr is NULL in first place.